service/get.go

```diff
@@ -81,3 +90,26 @@ func writeFile(name string, content []byte, log *log.Logger) error {
 	}
 	return nil
 }
+
+func prepareIndexFile(folder string, URL string, newChartHost string, log *log.Logger) error {
+	indexContent, err := ioutil.ReadFile(folder + "/downloaded-index.yaml")
+	if err != nil {
+		return err
+	}
+	content := string(indexContent)
+	if newChartHost != "" {
+		content = strings.Replace(content, URL, newChartHost, -1)
+	}
+
+	err = writeFile(folder+"/index.yaml", []byte(content), log)
+	if err != nil {
+		return err
+	}
+
+	err = os.RemoveAll(folder + "/downloaded-index.yaml")
+	if err != nil {
+		return err
+	}
+
+	return nil
```

Comment:
I would change the code to do something like (pseudo - code):

```
if newRootUrl != "" {
  old_contents = read_file
  new_contents = replace(old_contents, origRootUrl, newRootUrl)
  write_file(`index-download.yaml`, new_contents)
}

os.rename(`index-download.yaml`, `index.yaml`)
```

Just to avoid to read all the file in memory if there's no replacement to do.